### PR TITLE
Feat/symbion arm

### PR DESCRIPTION
**Pull request by degrigis**
Supporting ARM

### CODE REVIEW

**Comment by ltfish** [comment]
Sync https://github.com/angr/archinfo/pull/67.

File: tests/test_concrete_not_packed_elf32_arm.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import angr
+import avatar2
+import claripy
+import nose
+import os
+
+from angr_targets import AvatarGDBConcreteTarget
+
+GDB_SERVER_IP = '192.168.60.3'
```
Comment (ltfish) [inline tests/test_concrete_not_packed_elf32_arm.py]: How will this work on CI?

Comment (degrigis) [reply tests/test_concrete_not_packed_elf32_arm.py]: I've removed the hardcoded IP, but anyway, this test won't actually run on CI.

File: tests/test_concrete_not_packed_elf32_arm.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+def solv_concrete_engine_linux_arm(p, entry_state):
+    new_concrete_state = execute_concretly(p, entry_state, BINARY_DECISION_ADDRESS, [])
+
+    arg0 = claripy.BVS('arg0', 5 * 32)
+    symbolic_buffer_address = new_concrete_state.regs.r3
+    new_concrete_state.memory.store(symbolic_buffer_address, arg0)
+    simgr = p.factory.simgr(new_concrete_state)
+
+    print("Symbolically executing BINARY to find dropping of second stage [ address:  " + hex(DROP_STAGE2_V1) + " ]")
```
Comment (rhelmot) [inline tests/test_concrete_not_packed_elf32_arm.py]: let's not have print statements in our tests - commenting them out is fine if you expect to have to debug this in the future but it'll clutter the nose output

**Comment by ltfish** [comment]
- CI fails.
- If the test case does not work on the CI, can you rename the file so our CI script does not automatically pick it up?

**Comment by ltfish** [comment]
Linting fails. Fix linting and I'll hit merge!

**Pull request merged**